wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -530,6 +530,21 @@ impl PhysicalDeviceFeatures {
             supports_bgra8unorm_storage(instance, phd, caps.device_api_version),
         );
 
+        // https://github.com/KhronosGroup/MoltenVK/issues/440
+        #[cfg(not(any(target_os = "macos", target_os = "ios")))]
+        features.set(
+            F::TEXTURE_FORMAT_NV12,
+            supports_format(
```

Comment:
I don't think this is enough.

VK_FORMAT_G8_B8R8_2PLANE_420_UNORM(_KHR) was added in VK_KHR_sampler_ycbcr_conversion or Vulkan 1.1. This should check for the extension or Vulkan version and reject the format if both are false.

Reply:
> Validation Error: [ VUID-VkImageCreateInfo-pNext-06722 ] Object 0: handle = 0x1400bbe18, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x1a04b807 | vkCreateImage(): VkImageFormatListCreateInfo::pViewFormats[0] (VK_FORMAT_R8_UNORM) and VkImageCreateInfo::format (VK_FORMAT_G8_B8R8_2PLANE_420_UNORM) are not compatible. The Vulkan spec states: If a VkImageFormatListCreateInfo structure was included in the pNext chain and VkImageFormatListCreateInfo::viewFormatCount is not zero, then each format in VkImageFormatListCreateInfo::pViewFormats must either be compatible with the format as described in the compatibility table or, if flags contains VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT, be an uncompressed format that is size-compatible with format (https://vulkan.lunarg.com/doc/view/1.3.250.1/mac/1.3-extensions/vkspec.html#VUID-VkImageCreateInfo-pNext-06722)

Ci always failed on macOS, but I can create nv12 texture on my windows pc.

Reply:
Probably wasn't clear, you need to do the extension check/version check AND supports_format

Technically the format could be unsupported in Vulkan 1.1 but it is a known format.

Reply:
> Probably wasn't clear, you need to do the extension check/version check AND supports_format
> 
> Technically the format could be unsupported in Vulkan 1.1 but it is a known format.

I do the extension check/version check and supports_forma, but got the same error, I think it's a bug of MoltenVK.